Modelica/Mechanics/Translational/Components/Brake.mo

```diff
@@ -1,43 +1,62 @@
 within Modelica.Mechanics.Translational.Components;
 model Brake "Brake based on Coulomb friction"
-
   extends Modelica.Mechanics.Translational.Interfaces.PartialElementaryTwoFlangesAndSupport2;
   extends Modelica.Thermal.HeatTransfer.Interfaces.PartialElementaryConditionalHeatPortWithoutT;
+
+  import Modelica.Blocks.Types.{ExternalCombiTable1D, Extrapolation, Smoothness};
+  import Modelica.Blocks.Tables.Internal.{getTable1DValue, getTable1DValueNoDer, getTable1DValueNoDer2};
+
   parameter Real mu_pos[:, 2]=[0, 0.5]
     "Positive sliding friction coefficient [-] as function of v_rel [m/s] (v_rel>=0)";
+  parameter Smoothness smoothness = Smoothness.LinearSegments
+    "Smoothness of interpolation in mu_pos table";
   parameter Real peak(final min=1) = 1
     "Peak for maximum value of mu at w==0 (mu0_max = peak*mu_pos[1,2])";
   parameter Real cgeo(final min=0) = 1
     "Geometry constant containing friction distribution assumption";
   parameter SI.Force fn_max(final min=0, start=1) "Maximum normal force";
+
   extends Translational.Interfaces.PartialFriction;
 
   SI.Position s "Absolute position of flange_a and of flange_b";
   SI.Force f "Brake friction force";
   SI.Velocity v "Absolute velocity of flange_a and flange_b";
   SI.Acceleration a "Absolute acceleration of flange_a and flange_b";
 
-  Real mu0 "Friction coefficient for v=0 and forward sliding";
+  Real mu "Friction coefficient";
   SI.Force fn "Normal force (=fn_max*f_normalized)";
-
-  // Constant auxiliary variable
   Modelica.Blocks.Interfaces.RealInput f_normalized
     "Normalized force signal 0..1 (normal force = fn_max*f_normalized; brake is active if > 0)"
     annotation (Placement(transformation(
         origin={0,110},
         extent={{20,-20},{-20,20}},
         rotation=90)));
+
+protected
+  final parameter ExternalCombiTable1D tableID = ExternalCombiTable1D(
+    tableName="NoName",
+    fileName="NoName",
+    table=mu_pos,
+    columns={2},
+    smoothness=smoothness,
+    extrapolation=Extrapolation.LastTwoPoints,
+    verboseRead=false) "External table object for sliding friction coefficient";
+  final parameter Real mu0=
+    if     smoothness == Smoothness.ConstantSegments then getTable1DValueNoDer(tableID, 1, 0)
+    elseif smoothness == Smoothness.LinearSegments   then getTable1DValueNoDer2(tableID, 1, 0)
+    else                                                  getTable1DValue(tableID, 1, 0)
+    "Friction coefficient for w=0 and forward sliding";
+
+  Real table_signs[2]
+    "Signs for sliding friction coefficient table interpolation: [sign for w_rel, sign for mu]";
```

Comment:
There is no `w_rel` in Brake.

Reply:
Fix with 850a094b505c4e4737beccd24ef8a4ce6f85f114